components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -1651,6 +1684,37 @@ private void parseSupportedResponseTypesConfig(OMElement oauthConfigElem) {
         }
     }
 
+    private void parseDefaultSupportedResponseTypeConfig(List<String> supportedResponseTypes) {
+        if (!supportedResponseTypes.contains(ResponseType.CODE.toString())) {
```

Comment:
Can you explain the logic here?
why are we checking !supportedResponseTypes.contains(ResponseType.CODE.toString() condition?